torch/nn/modules/module.py

```diff
@@ -776,7 +921,7 @@ def _maybe_warn_non_full_backward_hook(self, inputs, result, grad_fn):
             inputs = (inputs,)
 
         # At this point we are sure that inputs and result are tuple of Tensors
-        out_grad_fn = set([r.grad_fn for r in result if r.grad_fn is not None])
+        out_grad_fn = {r.grad_fn for r in result if r.grad_fn is not None}
```

Comment:
Why is a dict better than a set here?

Reply:
This is a set comprehension, so it's the same thing as what we had before. I changed it because `flake8` doesn't like the [set constructor with a list comprehension](https://pypi.org/project/flake8-comprehensions/) for creating the temporary list

Reply:
Can we use the explicit set constructor with the bare generator expression:

```
set(r.grad_fn for r in result if r.grad_fn is not None)
```

IMO the fact that python uses the same tokens for both set and dict literals is confusing and making things more explicit can improve readability